hazelcast/src/main/java/com/hazelcast/query/impl/BaseIndexStore.java

```diff
@@ -48,64 +49,69 @@
         }
     }
 
-    abstract void newIndexInternal(Comparable newValue, QueryableEntry record);
+    abstract Object newIndexInternal(Comparable newValue, QueryableEntry record);
 
-    abstract void removeIndexInternal(Comparable oldValue, Data indexKey);
+    abstract Object removeIndexInternal(Comparable oldValue, Data indexKey);
```

Comment:
Maybe a javaDoc to explain new returned values of `newIndexInternal` and `removeIndexInternal`?

Reply:
fixed